### ISSUE TITLE
How to resize without cropping and distortion?

### ISSUE THREAD

**Issue by twarkie** [h2non/imaginary#135]
Hi!

I'm pretty new to this project but it's the best service I've found so far and I'm already using it in production.

My goal is to resize remote images on-the-fly to a set with and height while keeping the ratio. I'm used to the width and height being max dimensions, ie. if the image width is smaller than the provided max width, it's ok to scale it down while maintaining the max height. I can't make this work with Imaginary. This is what I have tried so far:

Source image is 500x1000px. Expected result is to end up with a resized image of 100x200px.

/resize?width=200&height=200&url=<remote url>
Result: crops image to 200x200

/resize?width=200&height=200&nocrop=true&url=<remote url>
Result: Image resized correctly, but it's added on a black background padded to 200x200px.

/resize?width=200&height=200&nocrop=true&extend=background&background=255,255,255&url=<remote url>
Result: My current solution, resizes the image and adds it on a white background padded to 200x200

How can I resize the image to 100x200px without cropping or distorting the image? I still need to pass the max width of 200 because that's what the site can handle.

Thank you!

**Comment by Dynom** [comment]
I don't think this is currently possible. Possibly a `max-width|height` ought to be introduced to support this behaviour, otherwise the meaning will become ambiguous. Alternatively a `maintain aspect ratio` property could be introduced, to better specify what `width|height` refers to.

As a work-around, you could decide the ratio on forehand and include only the parameter of the longest side, e.g.:

```go
var url string
if width / height <= 0.5 {
  url = "/resize?width=200&url=.."
} else {
  url = "/resize?height=200&url=.."
}
```
_So you only ever specify a width, or a height._

However, if you already know the dimensions up-front, you can just calculate the dimensions and specify it regardless, so it might not be an acceptable work-around for you.

**Comment by iwan41** [comment]
@twarkie maybe thumbnail is what you need.
You simply send only one dimension and another one will be counted. So in your case should be:

/resize?width=200&url=
